packages/app-transfer/src/Transfer.tsx

```diff
@@ -117,11 +142,62 @@ class Transfer extends React.PureComponent<Props, State> {
         amount,
         extrinsic,
         hasAvailable,
+        maxBalance,
         recipientId
       };
     });
   }
 
+  private setMaxBalance = () => {
+    const { api, balances_fees = ZERO_FEES } = this.props;
+    const { accountId, recipientId } = this.state;
+
+    if (!accountId || !recipientId) {
+      return;
+    }
+
+    void api.query.system.accountNonce(accountId, (accountNonce) => {
+
+      const { transferFee, transactionBaseFee, transactionByteFee, creationFee } = balances_fees;
+
+      void api.derive.balances.votingBalance(accountId, ({ freeBalance: senderBalance }) => {
+        void api.derive.balances.votingBalance(recipientId, ({ freeBalance: recipientBalance }) => {
+
+          let prevMax = new BN(0);
+          let maxBalance = new BN(1);
+          let extrinsic;
+
+          while (!prevMax.eq(maxBalance)) {
+            prevMax = maxBalance;
+
+            extrinsic = accountId && recipientId
+              ? api.tx.balances.transfer(recipientId, prevMax)
+              : null;
+
+            const txLength = SIGNATURE_SIZE + accountNonce.length + (
+              extrinsic
+                ? extrinsic.encodedLength
+                : 0
+            );
+
+            const fees = transactionBaseFee
+              .add(transactionByteFee.muln(txLength))
+              .add(transferFee)
+              .add(senderBalance.isZero() ? creationFee : ZERO)
+              .add(recipientBalance.isZero() ? creationFee : ZERO);
+
```

Comment:
We need to take into account the locked balance (available from balances.locks). In my tests so far, the max of a stash account fully bonded was wrong because the account is actually unable to transfer any fund. I've done something similar (need review) here: https://github.com/polkadot-js/apps/blob/da179c7550f7b210fa3610ff37a8375f72896900/packages/ui-reactive/src/Available.tsx